src/loaders/png/tvgPngLoader.cpp

```diff
@@ -180,10 +181,14 @@ unique_ptr<Surface> PngLoader::bitmap()
 
 void PngLoader::run(unsigned tid)
 {
+    if (image) {
```

Comment:
just a question - how can we achieve `image != null` here?

Reply:
if user try to read the data twice? 
open() -> read() -> open() -> read();

Reply:
Null checking is not quite necessary. Either case it's trivial, you can refactor the code if you want. :-)